### PR TITLE
Make sysroot use `RUST_SRC_PATH` if set

### PR DESCRIPTION
**Pull request by bkchr**

### CODE REVIEW

File: crates/ra_project_model/src/sysroot.rs [hunk 1]
```diff
@@ -23,6 +24,23 @@ struct SysrootCrateData {
     deps: Vec<SysrootCrate>,
 }
 
+fn try_find_src_path(cargo_toml: &Path) -> Result<PathBuf> {
+    if let Ok(path) = env::var("RUST_SRC_PATH") {
```
Comment (matklad) [inline crates/ra_project_model/src/sysroot.rs]: Is `RUST_SRC_PATH` a know env variable which is used elsewhere? If not, let's prefix it wiith `RA_`

File: crates/ra_project_model/src/sysroot.rs [hunk 2]
```diff
@@ -23,6 +24,23 @@ struct SysrootCrateData {
     deps: Vec<SysrootCrate>,
 }
 
+fn try_find_src_path(cargo_toml: &Path) -> Result<PathBuf> {
```
Comment (matklad) [inline crates/ra_project_model/src/sysroot.rs]: Let's move this fucntion *after* `impl Sysroot`

File: crates/ra_project_model/src/sysroot.rs (hunk 1 above)
Comment (matklad) [reply crates/ra_project_model/src/sysroot.rs]: Should we perhaps make this a config option? I *think* env var makes slightly sense in this case, as this is something you probably shouldn't configure, unless you know what you are doing, and, in that case, you probably want to easily share this for all instaces of rust-analyzer

Comment (bkchr) [reply crates/ra_project_model/src/sysroot.rs]: > Is `RUST_SRC_PATH` a know env variable which is used elsewhere? If not, let's prefix it wiith `RA_`

`RUST_SRC_PATH` is a known variable in the rust ecosystem. Racer uses it and other projects seems to use it too.

I would like to stick to an env variable. In my setup I have multiple rust versions, mostly one rust version per project. So, I can not share the src path between multiple projects.

Comment (matklad) [reply crates/ra_project_model/src/sysroot.rs]: cc @vlad20012, might make sense to support `RUST_SRC_PATH` in IntelliJ as well.

@bkchr let's then just mention this env var in the docs!

**Comment by bkchr** [comment]
@matklad I added it to the docs.

**Comment by matklad** [comment]
bors r+

Thanks!

**Pull request merged**